django_components/template_loader.py

```diff
@@ -2,34 +2,84 @@
 Template loader that loads templates from each Django app's "components" directory.
 """
 
+import os
 from pathlib import Path
+from typing import Set
 
+from django.apps import apps
 from django.conf import settings
 from django.template.loaders.filesystem import Loader as FilesystemLoader
-from django.template.utils import get_app_template_dirs
+
+from django_components.logger import logger
+
+
+# Same as `Path.is_relative_to`, defined as standalone function because `Path.is_relative_to`
+# is marked for deprecation.
+def path_is_relative_to(child_path: str, parent_path: str):
+    # If the relative path doesn't start with `..`, then child is descendant of parent
+    # See https://stackoverflow.com/a/7288073/9788634
+    rel_path = os.path.relpath(child_path, parent_path)
+    return not rel_path.startswith("..")
 
 
 class Loader(FilesystemLoader):
     def get_dirs(self):
-        component_dir = "components"
-        directories = set(get_app_template_dirs(component_dir))
```

Comment:
Note: I've replaced `get_app_template_dirs` with [the loop on line 46](https://github.com/EmilStenstrom/django-components/pull/395/files#diff-c6c9a460aa194de49542b7c8e241152e7727fd8d1ea49f77c3437b0373427d24R46), which is based on the implementation of `get_app_template_dirs`